### ISSUE TITLE
Scale down deletes two nodes when in Provisioning state

### ISSUE THREAD

**Issue by sowmyav27** [rancher/rancher#36502]
**Rancher Server Setup**
- Rancher version: 2.6-head commit id: `11a7451`
- Installation option (Docker install/Helm Chart): docker install

**Information about the Cluster**
- Kubernetes version: `v1.21.9+rke2r1`
- Cluster Type (Local/Downstream): DO Node driver cluster - 1 etcd, 1 cp, 1 worker nodes

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom) Standard User

   
**Describe the bug**
Scale down deletes two nodes when in Provisioning state

**To Reproduce**

- Deploy an RKE2 cluster - 1 etcd, 1 cp and 1 worker node - DO Node driver cluster
- Once the cluster comes up Active, Scale Up the worker nodes by 2
- The nodes will be in provisioning state.
- Scale down the Worker nodes, when in provisioning state by clicking on `-` in the worker pool.

**Result**
Both the nodes that are in provisioning state get deleted and a new one gets provisioning.

**Expected Result**
Only one of the two in provisioning state must get deleted when user scales by one.

**Rancher Logs:**
```
2022/02/13 21:27:42 [INFO] rkecluster fleet-default/sowmya-scale-down: provisioning control plane node(s) sowmya-scale-down-pool2-67d5fc557c-bs8rp
2022/02/13 21:27:43 [INFO] rkecluster fleet-default/sowmya-scale-down: provisioning control plane node(s) sowmya-scale-down-pool2-67d5fc557c-bs8rp: waiting for plan to be applied
2022/02/13 21:27:43 [INFO] rkecluster fleet-default/sowmya-scale-down: provisioning control plane node(s) sowmya-scale-down-pool2-67d5fc557c-bs8rp: waiting for plan to be applied
2022/02/13 21:27:43 [INFO] rkecluster fleet-default/sowmya-scale-down: provisioning control plane node(s) sowmya-scale-down-pool2-67d5fc557c-bs8rp: waiting for plan to be applied
2022/02/13 21:27:43 [INFO] rkecluster fleet-default/sowmya-scale-down: provisioning control plane node(s) sowmya-scale-down-pool2-67d5fc557c-bs8rp: waiting for plan to be applied
2022/02/13 21:27:44 [INFO] rkecluster fleet-default/sowmya-scale-down: provisioning control plane node(s) sowmya-scale-down-pool2-67d5fc557c-bs8rp: waiting for plan to be applied
2022/02/13 21:27:46 [ERROR] error syncing 'fleet-default/sowmya-scale-down-pool3-0c939a18-6c9hv': handler machine-provision: resource name may not be empty, requeuing
2022/02/13 21:27:52 [INFO] rkecluster fleet-default/sowmya-scale-down: provisioning control plane node(s) sowmya-scale-down-pool2-67d5fc557c-bs8rp: waiting for plan to be applied
2022/02/13 21:27:52 [INFO] rkecluster fleet-default/sowmya-scale-down: provisioning worker node(s) sowmya-scale-down-pool3-5bf64958cf-8v4xs,sowmya-scale-down-pool3-5bf64958cf-w4x4k
2022/02/13 21:27:52 [INFO] rkecluster fleet-default/sowmya-scale-down: provisioning worker node(s) sowmya-scale-down-pool3-5bf64958cf-8v4xs,sowmya-scale-down-pool3-5bf64958cf-w4x4k
2022/02/13 21:27:52 [INFO] rkecluster fleet-default/sowmya-scale-down: provisioning worker node(s) sowmya-scale-down-pool3-5bf64958cf-8v4xs,sowmya-scale-down-pool3-5bf64958cf-w4x4k
2022/02/13 21:27:52 [INFO] rkecluster fleet-default/sowmya-scale-down: provisioning worker node(s) sowmya-scale-down-pool3-5bf64958cf-8v4xs,sowmya-scale-down-pool3-5bf64958cf-w4x4k
2022/02/13 21:27:53 [INFO] rate limited mcc-bundle(map[objectset.rio.cattle.io/hash:ae442cea0964fa218c862790e5ea113403d66446]) 1.209251919s
2022/02/13 21:28:10 [ERROR] Error during subscribe websocket: close sent
2022/02/13 21:28:13 [INFO] rkecluster fleet-default/sowmya-scale-down: provisioning control plane node(s) sowmya-scale-down-pool2-67d5fc557c-bs8rp
2022/02/13 21:28:13 [INFO] rkecluster fleet-default/sowmya-scale-down: provisioning control plane node(s) sowmya-scale-down-pool2-67d5fc557c-bs8rp: waiting for plan to be applied
2022/02/13 21:28:13 [INFO] rkecluster fleet-default/sowmya-scale-down: provisioning control plane node(s) sowmya-scale-down-pool2-67d5fc557c-bs8rp: waiting for plan to be applied
2022/02/13 21:28:14 [INFO] rkecluster fleet-default/sowmya-scale-down: provisioning control plane node(s) sowmya-scale-down-pool2-67d5fc557c-bs8rp: waiting for plan to be applied
2022/02/13 21:28:14 [INFO] rkecluster fleet-default/sowmya-scale-down: provisioning control plane node(s) sowmya-scale-down-pool2-67d5fc557c-bs8rp: waiting for plan to be applied
2022/02/13 21:28:14 [INFO] rkecluster fleet-default/sowmya-scale-down: provisioning control plane node(s) sowmya-scale-down-pool2-67d5fc557c-bs8rp: waiting for plan to be applied
2022/02/13 21:28:14 [ERROR] Reconciler error: Operation cannot be fulfilled on machinedeployments.cluster.x-k8s.io "sowmya-scale-down-pool3": the object has been modified; please apply your changes to the latest version and try again
2022/02/13 21:28:14 [ERROR] error syncing 'fleet-default/sowmya-scale-down-pool3-0c939a18-6c9hv': handler machine-provision-remove: cannot delete machine sowmya-scale-down-pool3-0c939a18-6c9hv because create job has not finished, requeuing
2022/02/13 21:28:14 [ERROR] error syncing 'fleet-default/sowmya-scale-down-pool3-0c939a18-6c9hv': handler machine-provision-remove: cannot delete machine sowmya-scale-down-pool3-0c939a18-6c9hv because create job has not finished, requeuing
2022/02/13 21:28:14 [ERROR] error syncing 'fleet-default/sowmya-scale-down-pool3-0c939a18-6c9hv': handler machine-provision-remove: cannot delete machine sowmya-scale-down-pool3-0c939a18-6c9hv because create job has not finished, requeuing
2022/02/13 21:28:14 [ERROR] error syncing 'fleet-default/sowmya-scale-down-pool3-0c939a18-6c9hv': handler machine-provision-remove: cannot delete machine sowmya-scale-down-pool3-0c939a18-6c9hv because create job has not finished, requeuing
2022/02/13 21:28:14 [ERROR] error syncing 'fleet-default/sowmya-scale-down-pool3-0c939a18-6c9hv': handler machine-provision-remove: cannot delete machine sowmya-scale-down-pool3-0c939a18-6c9hv because create job has not finished, requeuing
2022/02/13 21:28:15 [ERROR] error syncing 'fleet-default/sowmya-scale-down-pool3-0c939a18-6c9hv': handler machine-provision-remove: cannot delete machine sowmya-scale-down-pool3-0c939a18-6c9hv because create job has not finished, requeuing
2022/02/13 21:28:15 [ERROR] error syncing 'fleet-default/sowmya-scale-down-pool3-0c939a18-sqgbc': handler machine-provision-remove: DesiredSet - Replace Wait batch/v1, Kind=Job fleet-default/sowmya-scale-down-pool3-0c939a18-sqgbc-machine-provision for  fleet-default/sowmya-scale-down-pool3-0c939a18-sqgbc, requeuing
2022/02/13 21:28:15 [ERROR] error syncing 'fleet-default/sowmya-scale-down-pool3-0c939a18-6c9hv': handler machine-provision-remove: cannot delete machine sowmya-scale-down-pool3-0c939a18-6c9hv because create job has not finished, requeuing
2022/02/13 21:28:15 [ERROR] error syncing 'fleet-default/sowmya-scale-down-pool3-0c939a18-6c9hv': handler machine-provision-remove: cannot delete machine sowmya-scale-down-pool3-0c939a18-6c9hv because create job has not finished, requeuing
2022/02/13 21:28:16 [ERROR] error syncing 'fleet-default/sowmya-scale-down-pool3-0c939a18-6c9hv': handler machine-provision-remove: cannot delete machine sowmya-scale-down-pool3-0c939a18-6c9hv because create job has not finished, requeuing
2022/02/13 21:28:17 [ERROR] error syncing 'fleet-default/sowmya-scale-down-pool3-0c939a18-6c9hv': handler machine-provision-remove: cannot delete machine sowmya-scale-down-pool3-0c939a18-6c9hv because create job has not finished, requeuing
2022/02/13 21:28:19 [ERROR] error syncing 'fleet-default/sowmya-scale-down-bootstrap-template-d8dp9': handler rke-machine: failed to delete fleet-default/sowmya-scale-down-bootstrap-template-d8dp9-machine-plan /v1, Kind=ServiceAccount for rke-machine fleet-default/sowmya-scale-down-bootstrap-template-d8dp9: serviceaccounts "sowmya-scale-down-bootstrap-template-d8dp9-machine-plan" not found, requeuing
2022/02/13 21:28:19 [ERROR] Reconciler error: machines.cluster.x-k8s.io "sowmya-scale-down-pool3-5bf64958cf-8v4xs" not found
2022/02/13 21:28:20 [ERROR] error syncing 'fleet-default/sowmya-scale-down-pool3-0c939a18-6c9hv': handler machine-provision-remove: cannot delete machine sowmya-scale-down-pool3-0c939a18-6c9hv because create job has not finished, requeuing
```

**Comment by paynejacob** [comment]
@thedadams will create an upstream issue regarding this behavior.  This is not a bug necessarily but it may not be intentional behavior and we should bring it up with the upstream party.

**Comment by thedadams** [comment]
Upstream issue: https://github.com/kubernetes-sigs/cluster-api/issues/6299

**Comment by snasovich** [comment]
Moved to blocked as we're waiting for upstream on this one.

**Comment by thedadams** [comment]
Upstream PR has been merged: https://github.com/kubernetes-sigs/cluster-api/pull/6300

**Comment by Oats87** [comment]
Should be resolved as of: https://github.com/rancher/rancher/pull/38292

**Comment by snasovich** [comment]
This should be fixed indirectly due to us updating CAPI dependency to 1.2.0.

**Comment by markusewalker** [comment]
Verified that this is addressed in `v2.6-head` https://github.com/rancher/rancher/commit/edf595e60b2a6da340347499bd5b3ac7bd1f1ff4.

**ENVIRONMENT DETAILS**
- Rancher Install: Helm
- Rancher Version: `v2.6-head` https://github.com/rancher/rancher/commit/edf595e60b2a6da340347499bd5b3ac7bd1f1ff4
- Browser: Chrome

**TEST RESULT**
PASS

**VERIFICATION STEPS**
1. Setup Rancher and navigate to the Ui in a browser.
2. Created a standard user and login as that user.
3. Provisioned a downstream RKE2 node driver cluster with the following configuration: 3 etcd, 2 cp, 3 workers.
4. Verified that the cluster and nodes are in `Active` state in Cluster Management and Cluster Explorer.
5. In Cluster Management, clicked the provisioned cluster > Machine Pools.
6. In the worker pool, scaled up the nodes by two; verified that there are now 5 worker nodes and the cluster is in a state of `Provisioning`.
7. While the cluster is in `Provisioning`, scaled down the worker pool; verified that you only have one node removed and have 4 worker nodes.
8. Navigated to the Rancher logs and verified that there are no error messages pertaining to scaling down the worker nodes.
